### PR TITLE
Iana/home page

### PR DESCRIPTION
**Pull request by alekseevaiana**
Home page logic for managing meals (without styling and storing data):
- meals and plan meals lists;
- create new meal;
- add/remove meal to plan;

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -25,7 +26,8 @@ function App() {
   };
 
   const handleAddMealBtn = () => {
-    const new_meal = { name: inputIngridientName, ingridients };
+    const lastId = mealsData[mealsData.length - 1].id;
+    const new_meal = { name: inputIngridientName, ingridients, id: lastId + 1 };
```
Comment (slonoed) [inline src/App.js]: Follow the same naming
```suggestion
    const newMeal = { name: inputIngridientName, ingridients, id: lastId + 1 };
```

File: src/components/Home.js [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+            path="/"
+            element={
+              <MealsList
+                meals={meals}
+                handlePlanBtn={handlePlanBtn}
+                handleOpenMealBtnClick={handleOpenMealBtnClick}
+              />
+            }
+          />
+          {console.log(mealsInPlan, "mealsInPlan")}
```
Comment (slonoed) [inline src/components/Home.js]: Remove log

File: src/components/IngridientsList.js [hunk 3]
```diff
@@ -0,0 +1,14 @@
+export default function IngridientsList({ meal }) {
+  return (
+    <>
+      {meal.ingridients.map((ingridient, index) => {
+        return (
+          <>
```
Comment (slonoed) [inline src/components/IngridientsList.js]: Elements should have `key` property

File: src/components/IngridientsList.js [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+            {ingridient}
+            {index != meal.ingridients.length - 1 ? ", " : ""}
```
Comment (slonoed) [inline src/components/IngridientsList.js]: Don't use `!=` and `==`. Instead always use `!==` and `===`

File: src/components/MealPage.js [hunk 5]
```diff
@@ -0,0 +1,24 @@
+import { Typography, Button } from "@mui/material";
+import { useParams } from "react-router-dom";
+import IngridientsList from "./IngridientsList";
+import { Link } from "react-router-dom";
+
+export default function MealPage({ mealsData }) {
+  const params = useParams();
+
+  const currentMeal = mealsData.find(
+    (meal) => meal.id === Number.parseInt(params.id)
```
Comment (slonoed) [inline src/components/MealPage.js]: 1. `Number.parseInt` can return `NaN`.  Consider handling it explicitly.
2. Don't run `parseInt` on every interaction. Precalculate it before calling `find`.

File: src/App.js (hunk 1 above)
Comment (alekseevaiana) [reply src/App.js]: ok

File: src/components/Home.js (hunk 2 above)
Comment (alekseevaiana) [reply src/components/Home.js]: ok

File: src/components/IngridientsList.js (hunk 3 above)
Comment (alekseevaiana) [reply src/components/IngridientsList.js]: ok

**Review by slonoed — APPROVED** [review]

**Pull request merged**